packages/embark-deploy-tracker/src/index.js

```diff
@@ -58,26 +58,37 @@ class DeployTracker {
     });
   }
 
-  setCurrentChain(cb) {
+  setCurrentChain(callback) {
     const self = this;
     if (this.chainConfig === false) {
       this.currentChain = {contracts: []};
-      return cb();
+      return callback();
     }
-    this.events.request("blockchain:block:byNumber", 0, function(err, block) {
-      if (err) {
-        return cb(err);
-      }
-      let chainId = block.hash;
 
-      if (self.chainConfig[chainId] === undefined) {
-        self.chainConfig[chainId] = {contracts: {}};
-      }
+    function getBlock(blockNum, cb) {
+      self.events.request("blockchain:block:byNumber", blockNum, (err, block) => {
+        if (err) {
+          return cb(err);
+        }
+        let chainId = block.hash;
+
+        if (self.chainConfig[chainId] === undefined) {
+          self.chainConfig[chainId] = {contracts: {}};
+        }
 
-      self.currentChain = self.chainConfig[chainId];
+        self.currentChain = self.chainConfig[chainId];
 
-      self.currentChain.name = self.env;
-      cb();
+        self.currentChain.name = self.env;
+        cb();
+      });
+    }
```

Comment:
This usage looks a little upside-down to me. `getBlock()` is changing state, while it probably should just emit the block that has been requested. 

We probably rather want to have `getBlock()` defined somewhere outside `setCurrentChain()`, then use it inside `setCurrentChain()` (which is actually done in L87) but have the callback of `getBlock()` do all the state changes.